broker/fragment/store_azure.go

```diff
@@ -183,17 +238,48 @@ func (a *azureBackend) azureClient(ep *url.URL) (cfg AzureStoreConfig, client pi
 		return
 	}
 
+	var credentials azblob.Credential
+
 	accountName := os.Getenv("AZURE_ACCOUNT_NAME")
-	accountKey := os.Getenv("AZURE_ACCOUNT_KEY")
-	credentials, err := azblob.NewSharedKeyCredential(accountName, accountKey)
-	if err != nil {
-		return
+	a.endpoint = fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+
+	if ep.Scheme == "azure" {
+		var accountKey = os.Getenv("AZURE_ACCOUNT_KEY")
+		sharedKeyCred, err := service.NewSharedKeyCredential(accountName, accountKey)
+		if err != nil {
+			return cfg, nil, err
+		}
+		serviceClient, err := service.NewClientWithSharedKeyCredential(a.endpoint, sharedKeyCred, &service.ClientOptions{})
+		if err != nil {
+			return cfg, nil, err
+		}
+		a.svcClient = *serviceClient
+		// azblob SharedKeyCredential is apparently different from service SharedKeyCredential
+		credentials, err = azblob.NewSharedKeyCredential(accountName, accountKey)
+		if err != nil {
+			return cfg, nil, err
+		}
+	} else if ep.Scheme == "azure-ad" {
+		// https://learn.microsoft.com/en-us/azure/developer/go/azure-sdk-authentication-service-principal?tabs=azure-cli#-option-1-authenticate-with-a-secret
```

Comment:
nit: See other comment about free-floating URLs like this.

Reply:
Added a comment